core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -1105,6 +1095,15 @@ private[spark] class AppStatusListener(
 
       cleanupCachedQuantiles(key)
     }
+
+    // Delete tasks for all stages in one pass, as deleting them for each stage individually is slow
```

Comment:
Hi @patrickbrownsync , can you explain more about why deleting them in the loop above is slower?  I don't quite understand it.